src/audio/signalinfo.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "audio/types.h"
+#include "util/assert.h"
+#include "util/macros.h"
+#include "util/optional.h"
+
+namespace mixxx {
+
+namespace audio {
+
+// Properties that characterize an uncompressed PCM audio signal.
+class SignalInfo final {
+    // Properties
+    PROPERTY_SET_BYVAL_GET_BYREF(ChannelCount, channelCount, ChannelCount)
+    PROPERTY_SET_BYVAL_GET_BYREF(SampleRate, sampleRate, SampleRate)
+    PROPERTY_SET_BYVAL_GET_BYREF(OptionalSampleLayout, sampleLayout, SampleLayout)
+
+  public:
+    constexpr SignalInfo() = default;
+    constexpr explicit SignalInfo(
+            OptionalSampleLayout sampleLayout)
+            : m_sampleLayout(sampleLayout) {
+    }
+    SignalInfo(
+            ChannelCount channelCount,
+            SampleRate sampleRate,
+            OptionalSampleLayout sampleLayout = std::nullopt)
+            : m_channelCount(channelCount),
+              m_sampleRate(sampleRate),
+              m_sampleLayout(sampleLayout) {
+    }
+    SignalInfo(SignalInfo&&) = default;
+    SignalInfo(const SignalInfo&) = default;
+    /*non-virtual*/ ~SignalInfo() = default;
+
+    constexpr bool isValid() const {
+        return getChannelCount().isValid() &&
+                getSampleLayout() &&
+                getSampleRate().isValid();
+    }
+
+    SignalInfo& operator=(SignalInfo&&) = default;
+    SignalInfo& operator=(const SignalInfo&) = default;
+
+    // Conversion: #samples / sample offset -> #frames / frame offset
```

Comment:
Not strictly necessary for this PR, but maybe it would be nice to have this new class be responsible for converting between real time (seconds/milliseconds) and sample/frame positions. It encapsulates the required information, right?